### PR TITLE
Added FixedPointIterationTargetDT

### LINKED ISSUE #27381: Add timestepper based on target fixed point iterations
### Motivation

The optimal time step size in terms of efficiency in a multi-physics simulation should consider the difficulty of the coupling, and there should be an alternative to just setting the maximum number of fixed point iterations, as this wastes a lot of time through failed time steps.

### Design

This would create a new time-stepper that adjusts the time step size based on the number of fixed point iterations in the previous time step: if greater than the target window, it decreases time step size, and if less than the target window, it increases the time step size.

### Impact

New object.

### PR DESCRIPTION
**Pull request by joshuahansel**
Closes #27381

### CODE REVIEW

**Comment by GiudGiud** [comment]
Instead of TargetDT, maybe AdaptiveDT, that way it follows the same naming convention as the other one?

**Comment by joshuahansel** [comment]
> Instead of TargetDT, maybe AdaptiveDT, that way it follows the same naming convention as the other one?
>
> alternatively you could fold this one into the IterationAdaptiveDT ?

I'll rename. I don't think anything else should go into IterationAdaptiveDT - I think it should be decomposed into multiple classes at some point.

File: framework/src/timesteppers/FixedPointIterationAdaptiveDT.C [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+registerMooseObject("MooseApp", FixedPointIterationAdaptiveDT);
+
+InputParameters
+FixedPointIterationAdaptiveDT::validParams()
+{
+  InputParameters params = TimeStepper::validParams();
+  params.addClassDescription(
+      "Computes time step size based on a target number of fixed point iterations");
+  params.addRequiredParam<Real>("dt_initial", "The initial time step size");
```
Comment (GiudGiud) [inline framework/src/timesteppers/FixedPointIterationAdaptiveDT.C:22]: use rangeChecked params for all the Real

File: test/tests/time_steppers/fixed_point_iteration_adaptive_dt/tests [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[Tests]
+  issues = '#27381'
+  design = 'source/timesteppers/FixedPointIterationAdaptiveDT.md'
+
+  [increase_dt]
+    type = CSVDiff
+    input = 'main.i'
+    csvdiff = 'increase_dt.csv'
+    requirement = "The system shall be able to increase the time step size when below the target number of fixed point iterations."
```
Comment (GiudGiud) [inline test/tests/time_steppers/fixed_point_iteration_adaptive_dt/tests:9]:
```suggestion
    requirement = "The system shall be able to increase the time step size when the simulation converges with less fixed point iterations than the target number."
```

File: test/tests/time_steppers/fixed_point_iteration_adaptive_dt/tests [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  []
+
+  [decrease_dt]
+    type = CSVDiff
+    input = 'main.i'
+    cli_args = 'Executioner/TimeStepper/target_iterations=4 Outputs/file_base=decrease_dt'
+    csvdiff = 'decrease_dt.csv'
+    requirement = "The system shall be able to decrease the time step size when above the target number of fixed point iterations."
```
Comment (GiudGiud) [inline test/tests/time_steppers/fixed_point_iteration_adaptive_dt/tests:17]:
```suggestion
 "The system shall be able to decrease the time step size when the simulation converges with more fixed point iterations than the target number."
```

File: test/tests/time_steppers/fixed_point_iteration_adaptive_dt/tests [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  []
+
+  [error_no_multiapps]
+    type = RunException
+    input = 'main.i'
+    cli_args = "MultiApps/sub/input_files=sub_with_stepper.i"
+    expect_err = 'This time stepper can only be used if there are MultiApps in the problem'
+    requirement = 'The system shall report an error if the fixed point target time stepper is used in an application with no sub-applications.'
```
Comment (GiudGiud) [inline test/tests/time_steppers/fixed_point_iteration_adaptive_dt/tests:25]:
```suggestion
    requirement = 'The system shall report an error if the fixed point target time stepper is used in an application with no child-applications and no fixed point iterations between applications.'
```

File: framework/src/timesteppers/FixedPointIterationAdaptiveDT.C [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+{
+}
+
+void
+FixedPointIterationAdaptiveDT::init()
+{
+  TimeStepper::init();
+
+  if (!_fe_problem.hasMultiApps())
+    mooseError("This time stepper can only be used if there are MultiApps in the problem.");
```
Comment (GiudGiud) [inline framework/src/timesteppers/FixedPointIterationAdaptiveDT.C:65]: need to check that fixed point iterations are in use, which the executioner would know about

**Review by GiudGiud — CHANGES_REQUESTED** [review]
pretty clever. I should have added something like that when working on SAM-Pgh FPs too it would have helped

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**